middleware/logger.js

```diff
@@ -0,0 +1,56 @@
+const shortId = require('shortid');
+
+/**
+ * Logs requests
+ */
+module.exports = (req, res, next) => {
+  const cleanup = () => {
+    res.removeListener('finish', log);
+    res.removeListener('close', abort);
+    res.removeListener('error', error);
+  };
+
+  const log = () => {
+    cleanup();
+    const logger = getLoggerByStatusCode(res.statusCode);
+    logger(
+      `${getPrefix(req)} ${res.statusCode} ${res.statusMessage} ${res.get(
+        'Content-Length',
+      ) || 0}b sent`,
+    );
+  };
+
+  const abort = () => {
+    cleanup();
+    console.warn(`${getPrefix(req)} Request aborted by client`);
+  };
+
+  const error = error => {
```

Comment:
```error``` is double defined, both as the constant function name and as your parameter. Rename the parameter to ```err``` or something
```suggestion
  const error = err => {
```
```js
const error = err => {
  cleanup();
  console(`${getPrefix(req)} Request pipeline error: ${err}`);
};
```